Composer/packages/client/src/recoilModel/dispatchers/botProjectFile.ts

```diff
@@ -14,7 +14,9 @@ import { rootBotProjectIdSelector } from '../selectors';
 import { setRootBotSettingState } from './setting';
 import { addSkillFiles, deleteSkillFiles } from './utils/skills';
 
-const urlRegex = /^http[s]?:\/\/\w+/;
+const isExternalLink = (url: string): boolean => {
```

Comment:
suggestion:
```js
const isExternalLink = (url: string): boolean => {
    const { origin } = location;
    const parsedUrl = new URL(url, origin);
    return parsedUrl.origin !== origin;
};
```
Note that in case the URL provided is invalid, this will throw.

My only concern is about protocols limiting to `http(s)`.

Also noted the implementation does not support relative protocol `//`. Though this is not supported in the code suggested as well, and I don't think it should be. Just raising the point in case there are known use-cases.